src/components/templates/PosterPage/components/ShareModal/ShareModal.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { FC, useState, useCallback } from "react";
+import { Modal } from "react-bootstrap";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { CopyToClipboard } from "react-copy-to-clipboard";
+import { faFacebook, faTwitter } from "@fortawesome/free-brands-svg-icons";
+import { DEFAULT_MAP_BACKGROUND } from "settings";
+import { WithId } from "utils/id";
+import { PosterPageVenue } from "types/venues";
+import classNames from "classnames";
+import { LinkButton } from "components/atoms/LinkButton";
+
+import "./ShareModal.scss";
+
+interface ConfirmationModalProps {
+  show: boolean;
+  onHide: () => void;
+  venue: WithId<PosterPageVenue>;
+}
+
+export const ShareModal: FC<ConfirmationModalProps> = ({
+  show,
+  onHide,
+  venue,
+}) => {
+  const url = `${process.env.REACT_APP_CODE_CHECK_URL}in/${venue.id}`;
+  const [isCopied, setIsCopied] = useState(false);
+
+  const handlerCopied = useCallback(() => {
+    setIsCopied(true);
+  }, []);
+
+  const linkText = isCopied ? "Link Copied" : "Copy Link";
+
+  const linkClasses = classNames("share-modal-container__link", {
+    "share-modal-container__link--copied": isCopied,
+  });
```

Comment:
`useShowHide`? and better naming eg. showCopiedText? Not sure about this tbh @0xdevalias

Reply:
I probably would personally. Removes the boilerplate of the `useState` + `useCallback`.

[OOS] `useShowHide` probably just needs a better name in general to make it more obvious in these sorts of cases.

Reply:
https://github.com/sparkletown/sparkle/pull/1396/commits/6ab7d04659ddfbdc2b675e3927a546b513eb0981